frontends/main/src/app/sitemaps/sitemap-index.xml/route.ts

```diff
@@ -0,0 +1,52 @@
+import { NextResponse } from "next/server"
+import invariant from "tiny-invariant"
+import * as resourceSitemap from "../resources/sitemap"
+import * as channelsSitemap from "../channels/sitemap"
+
+invariant(process.env.NEXT_PUBLIC_ORIGIN, "NEXT_PUBLIC_ORIGIN must be defined")
+const BASE_URL: string = process.env.NEXT_PUBLIC_ORIGIN
+
+export async function GET() {
```

Comment:
Oddly, NextJS has builtin support for generating multiple sitemaps via [`generateSitemaps`](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#generating-multiple-sitemaps). But no builtin support for a sitemap index. But it's pretty easy to make one manually.